frame/evm/src/lib.rs

```diff
@@ -629,11 +629,20 @@ impl<T: Trait> Module<T> {
 		);
 		executor.deposit(source, total_fee.saturating_sub(actual_fee));
 
+		let mut logs_result: Vec<Log> = Vec::new();
 		if apply_state {
 			let (values, logs) = executor.deconstruct();
-			backend.apply(values, logs, true);
+			let logs_data = logs.into_iter().map(|x| x ).collect::<Vec<_>>().clone();
```

Comment:
What's the reason that we skip handling logs return data when `apply_state` is false?

Reply:
No reason at all, just a matter if we want to return the log data on gas estimation or not. Do you think we should?

Reply:
Yes. I think if we're returning logs, we should always make sure the value is valid.